doc/source/notebooks/advanced/heteroskedastic.pct.py

```diff
@@ -214,6 +214,78 @@ def optimisation_step():
 
 model
 
+# %% [markdown]
+# ## The conditional output distribution
+#
+# Here we show how to get the conditional output distribution and plot samples from it. In order to plot the uncertainty associated with that, we also get the percentiles from the conditional output distribution.
+# Although the likelihood is Gaussian, the marginal posterior is not hence plotting with just the mean and variance would be misrepresentative of the real situation.
+
+# %%
+plot_distribution(X, Y, Ymean, Ystd)
+
+y_dist = model.conditional_y_dist(X)
+samples = y_dist.sample(10_000)
+
+# The folling is equivalent at doing:
+# y_lo_lo, y_lo, y_hi, y_hi_hi = np.quantile(samples, q=(0.025, 0.159, 0.841, 0.975), axis=0)
+# Note how, contrary to the binary classification case, here we get the percentiles directly from the
+# conditional output distribution
+y_lo_lo, y_lo, y_hi, y_hi_hi = y_dist.y_percentile(p=(2.5, 15.9, 84.1, 97.5), num_samples=10_000)
```

Comment:
Q: it's a bit annoying that `numpy` uses fractions of 1 and we use percentiles in our API. I guess this is because of TFP's API?